include/circt/Dialect/HW/HWOps.h

```diff
@@ -194,6 +195,10 @@ StringAttr getArgSym(Operation *op, unsigned i);
 /// argument.
 StringAttr getResultSym(Operation *op, unsigned i);
 
+/// Creates an i0-typed constant.
+Value getI0Constant(OpBuilder &builder, Location loc);
```

Comment:
Not sure these are appropriate here. Seems like the kinda thing which belongs in a specialized builder, but not free functions here. You're only using them in one place AFAICT, so I'd suggest dropping them.